src/pages/audio-rewards-page/store/sagas.test.ts

```diff
@@ -0,0 +1,331 @@
+            { error: FailureReason.COGNITO_FLOW }
+          ]
+        ])
+        // Assertions
+        .call.like({
+          fn: AudiusBackend.submitAndEvaluateAttestations,
+          args: [expectedRequestArgs]
+        })
+        .put(claimChallengeRewardWaitForRetry(testClaim))
+        .put(setVisibility({ modal: 'Cognito', visible: true }))
+        .not.put(
+          setVisibility({ modal: 'ChallengeRewardsExplainer', visible: false })
+        )
+        .silentRun(500)
+    )
+  })
+
+  it('should fail and not retry nor open models on user blocked', () => {
+    return (
+      expectSaga(saga)
+        .dispatch(
+          claimChallengeReward({
+            claim: testClaim,
+            retryOnFailure: true
+          })
+        )
+        .provide([
+          ...claimAsyncProvisions,
+          [
+            call.fn(AudiusBackend.submitAndEvaluateAttestations),
+            { error: FailureReason.BLOCKED }
+          ]
+        ])
+        // Assertions
+        .call.like({
+          fn: AudiusBackend.submitAndEvaluateAttestations,
+          args: [expectedRequestArgs]
+        })
+        .not.put(claimChallengeRewardWaitForRetry(testClaim))
+        .not.put(setVisibility({ modal: 'Cognito', visible: true }))
+        .not.put(
+          setVisibility({ modal: 'ChallengeRewardsExplainer', visible: false })
+        )
+        .put(claimChallengeRewardFailed())
+        .silentRun(5)
+    )
+  })
+
+  it('should update the audio balance and disbursement status on success', () => {
+    return (
+      expectSaga(saga)
+        .dispatch(
+          claimChallengeReward({
+            claim: testClaim,
+            retryOnFailure: true
+          })
+        )
+        .provide([
+          ...claimAsyncProvisions,
+          [call.fn(AudiusBackend.submitAndEvaluateAttestations), {}]
+        ])
+        // Assertions
+        .put(
+          increaseBalance({
+            amount: stringAudioToStringWei('10' as StringAudio)
+          })
+        )
+        .put(setUserChallengeDisbursed({ challengeId: 'connect-verified' }))
+        .put(claimChallengeRewardSucceeded())
+        .silentRun(5)
+    )
+  })
+
+  it('should wait until discovery marked the challenge as completed before submitting', () => {
+    return (
+      expectSaga(saga)
+        .dispatch(
+          claimChallengeReward({
+            claim: testClaim,
+            retryOnFailure: true
+          })
+        )
+        .provide([
+          [
+            select.like({
+              selector: getUserChallenge,
+              args: [{ challengeId: testUserChallenge.challenge_id }]
+            }),
+            { is_completed: false }
+          ]
+        ])
+        // Assertions
+        .not.call(AudiusBackend.submitAndEvaluateAttestations)
+        .not.put(setUserChallengeDisbursed({ challengeId: 'connect-verified' }))
+        .not.put(claimChallengeRewardSucceeded())
+        .silentRun(5)
```

Comment:
how did you pick these timeouts for silentRun?

Reply:
By reducing the timeout as much as I could without the tests failing 😅 probably a bit aggressive

Reply:
be careful that it's not faster on your machine and this works the same on CI. maybe worth having some default sensible longer timeouts? i doubt we will fail because of timeouts for any good reason so we should just let the test run

Reply:
We can't run without timeouts because take every and takeLatest etc never terminate. The test plan defaults to having some longish timeouts, though, so we could stick to those.

Alternatively, I could export the individual sagas to work around takeLatest etc but then I'm modifying the source for the test

Reply:
what if the default was just like 500 or something and we use that as a consistent config between tests?

Reply:
I think that is the default. I'll unset these manually configured timeouts and use that. We can tweak the default timeout separately (Imho that's wayy too long since it becomes the minimum time fo these tests)

Reply:
i think having different timing for each test where someone will have to know how to tweak them / stand the chance of arbitrarily failing will cost us more time than a 500ms/test run, but maybe 100 is reasonable. feel free to just merge as is. good discussion

Reply:
One last note: AFAIK timeouts don't cause test failures. It outputs a warning and some of your asserts might not get attempted. (This is bad)

Reply:
that seems really bad... sounds like picking a higher number is better even if the test is a little bit slower to avoid false negatives

Reply:
yeah. I should also not use silent so we can get those warnings. looks like default timeout is 250ms btw which is better: https://github.com/jfairbank/redux-saga-test-plan/blob/master/docs/integration-testing/timeout.md

Reply:
Well, actually those warnings aren't helpful, because I expect every test to timeout. I think maybe we lean towards exporting each individual generator we want to test, and that way we can avoid `takeEvery`/`takeLatest` and only test the generators they call underneath? Would help make the tests run faster, too.

Reply:
i'm ok with that. sometimes the nuance of takeLatest/takeEvery is important to capture, but generally I agree that testing the called saga is better. let's come up w/ a consistent naming pattern

```
doThing* () {
}

watchDoThing* () {
  yield takeEvery(actions.thing, doThing)
}
```